homeassistant/components/telegram_bot/__init__.py

```diff
@@ -377,33 +396,37 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Telegram bot component."""
     domain_config: list[dict[str, Any]] = config[DOMAIN]
 
-    if not domain_config:
-        return False
-
-    platforms = await async_get_loaded_integration(hass, DOMAIN).async_get_platforms(
-        {p_config[CONF_PLATFORM] for p_config in domain_config}
+    integration: Integration = async_get_loaded_integration(hass, DOMAIN)
+    modules: dict[str, ModuleType] = await integration.async_get_platforms(
+        [PLATFORM_BROADCAST, PLATFORM_POLLING, PLATFORM_WEBHOOKS]
     )
+    MODULES.update(modules)
 
-    for p_config in domain_config:
-        # Each platform config gets its own bot
-        bot = await hass.async_add_executor_job(initialize_bot, hass, p_config)
-        p_type: str = p_config[CONF_PLATFORM]
-
-        platform = platforms[p_type]
-
-        _LOGGER.debug("Setting up %s.%s", DOMAIN, p_type)
-        try:
-            receiver_service = await platform.async_setup_platform(hass, bot, p_config)
-            if receiver_service is False:
-                _LOGGER.error("Failed to initialize Telegram bot %s", p_type)
-                return False
-
-        except Exception:
-            _LOGGER.exception("Error setting up platform %s", p_type)
-            return False
-
-        notify_service = TelegramNotificationService(
-            hass, bot, p_config.get(CONF_ALLOWED_CHAT_IDS), p_config.get(ATTR_PARSER)
+    # import the last YAML config since existing behavior only works with the last config
+    if domain_config:
+        trusted_networks: list[IPv4Network] = domain_config[-1].get(
+            CONF_TRUSTED_NETWORKS, []
+        )
+        trusted_networks_str: list[str] = (
+            [str(trusted_network) for trusted_network in trusted_networks]
+            if trusted_networks
+            else []
+        )
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={CONF_SOURCE: SOURCE_IMPORT},
+                data={
+                    CONF_PLATFORM: domain_config[-1][CONF_PLATFORM],
+                    CONF_API_KEY: domain_config[-1][CONF_API_KEY],
+                    CONF_ALLOWED_CHAT_IDS: domain_config[-1][CONF_ALLOWED_CHAT_IDS],
+                    ATTR_PARSER: domain_config[-1].get(ATTR_PARSER),
```

Comment:
There's a default parser in the config schema.
```suggestion
                    ATTR_PARSER: domain_config[-1][ATTR_PARSER],
```